src/plugins/orangehrmOpenidAuthenticationPlugin/Controller/OpenIdConnectLoginController.php

```diff
@@ -44,40 +44,37 @@ class OpenIdConnectLoginController extends AbstractVueController implements Publ
      */
     public function getHomePageService(): HomePageService
     {
-        if (!$this->homePageService instanceof HomePageService) {
-            $this->homePageService = new HomePageService();
-        }
-        return $this->homePageService;
+        return $this->homePageService ??= new HomePageService();
     }
 
     /**
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     * @throws BadRequestException
      * @throws OpenIDConnectClientException
      */
-    public function handle(Request $request)
+    public function handle(Request $request): RedirectResponse
     {
-        $response = $this->getResponse();
-        $providerId = $request->request->get('providerId');
+        $providerId = $request->attributes->get('providerId');
+
         $provider = $this->getSocialMediaAuthenticationService()->getAuthProviderDao()
             ->getAuthProviderDetailsByProviderId($providerId);
+        if ($provider === null) {
+            throw new BadRequestException();
+        }
 
         $oidcClient = $this->getSocialMediaAuthenticationService()->initiateAuthentication(
             $provider,
-            self::SCOPE,
-            self::REDIRECT_URL
+            $this->getSocialMediaAuthenticationService()->getScope(),
+            $this->getSocialMediaAuthenticationService()->getRedirectURL()
         );
 
-        $email = $this->getSocialMediaAuthenticationService()->handleCallback($oidcClient);
-
+        $this->getAuthUser()->setAttribute(AuthUser::PROVIDER_ID, $provider->getId());
+        $oidcClient->authenticate();
 
-        if ($success) {
-            if ($this->getAuthUser()->isAuthenticated()) {
-                $homePagePath = $this->getHomePageService()->getHomePagePath();
-                return $this->redirect($homePagePath);
-            }
-            return parent::handle($request);
-        } else {
-            //handle error
-        }
-        return $response;
+        //redirect to consent always
+        $authUrl = $oidcClient->getGeneratedAuthUrl() . '&prompt=consent';
```

Comment:
```suggestion
        $authUrl = $oidcClient->getGeneratedAuthUrl();
```
can you  `&prompt=consent` check whether `prompt` is a standard param or is it google specific. But for google we need to keep `&prompt=consent`

Reply:
&prompt=consent using by other providers as well

Reply:
https://openid.net/specs/openid-connect-prompt-create-1_0.html